File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix A/CS/ServicesWithoutComponentsCS/App_Code/TransactionClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Text;
using System.EnterpriseServices;


/// <summary>
/// Summary description for TransactionClass
/// </summary>
public class TransactionClass
{
	public TransactionClass()
	{
		//
		// TODO: Add constructor logic here
		//
	}
    public string AddDetails(string prodName, string prodDescription, string
          prodType)
    {
        ServiceConfig config = new ServiceConfig();
        config.Transaction = TransactionOption.Required;
        ServiceDomain.Enter(config);
        try
        {
            Products prod = new Products();
            int ProdID = prod.AddProducts(prodName, prodDescription);
            ProductType pType = new ProductType();
            pType.AddProductType(ProdID, prodType);
        }
        catch
        {
            throw;
        }
        finally
        {
            ServiceDomain.Leave();
        }
        return "Data entered successfully";
    }

}

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix A/VB/ServicesWithoutComponentsVB/App_Code/TransactionClass.vb
Imports Microsoft.VisualBasic
Imports System.Data.SqlClient
Imports System.EnterpriseServices

Public Class TransactionClass
    Public Function AddDetails(ByVal prodName As String, ByVal prodDescription As String,
    ByVal prodType As String) As String
        Dim config As New ServiceConfig()
        config.Transaction = TransactionOption.Required
        ServiceDomain.Enter(config)
        Try
            Dim prod As New Products()
            Dim ProdID As Integer = prod.AddProducts(prodName, prodDescription)
            Dim pType As New ProductType()
            pType.AddProductType(ProdID, prodType)
        Catch
            Throw
        Finally
            ServiceDomain.Leave()
        End Try
        Return "Data entered successfully"
    End Function

End Class
